Changes to make it compile and run on powerpc64 Little Endian and Ubuntu 16.04

tzwenn opened the pull request.

tzwenn committed:
Littles fixes to let it compile on ppc64le
Also replace @bastih's backward-cpp with upstream.

tzwenn committed:
Make sure libcsv is compiled with g++ (discovered by @felixeberhardt)

felixeberhardt committed:
Fix issue #420: Failing test BufferedLoggerTests.log_test

felixeberhardt committed:
fix for failing TPCC tests which can get executed in a different order and need a cleared ResourceManager

felixeberhardt committed:
Add support for GCC 5

felixeberhardt committed:
Fix for Ubuntu 16.04(from @klauck): asm/msr.h does not include asm/msr-index.h anymore.

mrks commented on src/lib/helper/arch_compat.h:
```diff
@@ -0,0 +1,12 @@
+#endif
+
+#ifdef __powerpc64__
+#  define __builtin_ia32_sfence()
```
Maybe `atomic_thread_fence` would be the appropriate thing to use instead of the intrinsics? Not sure about the PPC execution model. Shouldn't keep us from pulling.

mrks replied:
Also, does this have to be a Macro or could it be an empty function?

tzwenn replied:
sfence is actually used only in the nvm-version but nowhere in hyrise. However I've made it consistent with the `PAUSE`-macro above.

mrks commented on src/lib/helper/arch_compat.h:
```diff
@@ -0,0 +1,12 @@
+
+#ifdef __SSE__
+#  include <immintrin.h>
+#  define PAUSE() _mm_pause()
```
Why is this inconsistent with how you work around sfence below?

mrks reviewed (changes requested):
I likez. The thing that I feel needs to be addressed is the march/mtune question. Maybe the other comments are worth looking at as well.

tzwenn committed:
Providing requested changes for pull request hyrise/hyrise#427:
	* Implemented sfence and mm_pause equivalents for ppc
	* Reintroduce -march for x64 (non-existend on power)

tzwenn commented on rules.mk:
```diff
@@ -193,7 +197,7 @@ LDFLAGS.release +=
 COMMON_FLAGS.debug += -O0
 COMMON_FLAGS.release += -O3
-COMMON_FLAGS += -ggdb -march=native -Wall -Wextra -Wno-attributes -Wno-unused-parameter -pthread $(COMMON_FLAGS.$(BLD))
+COMMON_FLAGS += -ggdb -mtune=native -Wall -Wextra -Wno-attributes -Wno-unused-parameter -pthread $(COMMON_FLAGS.$(BLD))
```
There is no [`-march` on Power ](https://gcc.gnu.org/onlinedocs/gcc/RS_002f6000-and-PowerPC-Options.html), but I'm using `-mcpu` now and reverted the Intel version.

tzwenn commented:
@bastih: Not really sure where we should point to otherwise. Just keep your repo public and around for the time being ;)
